components/survey_monkey/survey_monkey.app.mjs

```diff
@@ -73,25 +103,86 @@ export default {
 
       return axios($ || this, config);
     },
-    async getSurveys() {
-      return await this._makeRequest({
+    async _paginatedRequest(params) {
+      // https://api.surveymonkey.net/v3/docs?javascript#pagination
+      // using default SurveyMonkey pagination - 50 per page
+      const amountPerPage = 50;
+      const values = [];
+
+      const { path } = params;
+
+      let page = 1;
+      let response;
+
+      do {
+        response = await this._makeRequest({
+          ...params,
+          path: `${path}?page=${page}&per_page=${amountPerPage}`,
+        });
+
+        if (response.data) values.push(...response.data);
+        else throw new Error(response);
+      } while (page++ < Math.ceil(response.total / amountPerPage));
+
+      return values;
+    },
+    async getUserInfo({ $ }) {
+      return this._makeRequest({
+        method: "GET",
+        path: "/users/me",
+        $,
+      });
+    },
```

Comment:
You can simplify here in that way you don't have to set `$` symbol no each function. This same advice applies to all other functions with this approach.
```suggestion
    async getUserInfo(args = {}) {
      return this._makeRequest({
        method: "GET",
        path: "/users/me",
        ...args,
      });
    },
```

Reply:
That makes sense, but most of the methods use some of the `args` to assemble the URL, for example:
`async getResponseDetails({
      surveyId, responseId, $,
    }) {
      return this._makeRequest({
        method: "GET",
        path: `/surveys/${surveyId}/responses/${responseId}`,
        $,
      });
    },`
In this case, since surveyId and responseId will need to be destructured from the `args`, I prefer to destructure all of them right away; this also immediately shows what the method uses.

Do you think this change should be applied only for `getUserInfo` and `getSurveys` which require no parameters other than $? Or do you think there is a better approach to keep all the methods with the same pattern?

Reply:
Yeah in theses case what you can do is
```javascript
    async getResponseDetails({
      surveyId, responseId, ...args
    } = {}) {
      return this._makeRequest({
        method: "GET",
        path: `/surveys/${surveyId}/responses/${responseId}`,
        ...args,
      });
    },
```
in that way you don't deal with `$` here either

Reply:
Sure! That works well and is even more clear about which args are explicitly needed for this method.

Agreed and applied!